s4e-web/cypress/integration/map/map-layers.spec.ts

```diff
@@ -78,64 +68,85 @@ describe.skip('Map Layers', () => {
       response: this.geoserverResponse
     })
       .as('getCapabilities');
-    Layers
+    MapLayers
       .openManagementModal()
       .fillForm(label, `${geoserverUrl}?LAYERS=${layers}`);
     cy.wait('@getCapabilities');
 
-    Layers
+    MapLayers
       .activeLayersCountShouldBe(2)
       .allUrlLayersCountShouldBe(28);
 
-    GeneralModal
-      .closeAndChangeContext(Layers);
+    GeneralModal.closeModal();
   });
 
-  it('should add, hide and remove from panel', function () {
-    const label = 'Test';
-    const url = 'https://eumetview.eumetsat.int/geoserver/wms?LAYERS=eps:metop1_ir108';
-    const count = this.layerCount
-
-    Layers
-      .openManagementModal()
-      .addNew(label, url)
-    GeneralModal
-      .closeAndChangeContext(Layers)
-    Layers
-      .sidebarLayersCountShouldBe(count + 1)
-      .openManagementModal()
-      .toggleNthInPanelDisplay(count)
-    GeneralModal
-      .closeAndChangeContext(Layers);
-    Layers
-      .sidebarLayersCountShouldBe(count)
-      .openManagementModal()
-      .removeNthWithPermission(0)
-    ConfirmModal
-      .accept();
-  });
-
-  it('should display url errors', function () {
-    const label = 'Test';
-    const url = [
-      'https://test-page.pl/?SERVICE=unknown',
-      'REQUEST=unknown',
-      'LAYERS=unkn?own',
-      'STYLES=unkn?own',
-      'FORMAT=image/unknown',
-      'TRANSPARENT=none',
-      'VERSION=x.y.z',
-      'HEIGHT=none',
-      'WIDTH=none',
-      'CRS=unknown',
-      'BBOX=unknown,unknown,unknown,unknown'
-    ].join('&');
-
-    Layers
-      .openManagementModal()
-      .addNew(label, url)
-      .errorsCountShouldBe(6)
-      .changeContextTo(GeneralModal)
-      .closeAndChangeContext(Layers);
+  context("Configuration layers for logged in user", () => {
+
+    beforeEach(function () {
+      cy.visit('/login');
+      Login.loginAs(this.zkMember);
+      MapLayers.pageObject.getLayerCount();
+    });
+
+    it('should display selected layer', () => {
+      MapLayers
+        .selectNthSidebarLayer(0)
+        .activeLayersCountShouldBe(1)
+        .unselectNthSidebarLayer(0)
+        .activeLayersCountShouldBe(0);
+    });
+
+    it('should add, hide and remove layer from panel', function () {
+      const label = 'Test';
+      const url = 'https://eumetview.eumetsat.int/geoserver/wms?LAYERS=eps:metop1_ir108';
```

Comment:
I wouldn't depend on an external service. Maybe you could use or geoserver instance instead, i.e. the one set up for e2e tests. The url would be similar to `https://localhost:4202/wms?LAYERS=development:108m`. Maybe you could also somehow extract the baseUrl from cypress env (the baseUrl set in cypress.json, or in e2e gh-actions job)?

Reply:
I would be better to use mocked XML response

```
const label = 'Test';
    const geoserverUrl = '/test-wms';
    const layers = [
      'development:opad_h05_12h',
      'development:rgb24_micro'
    ].join(',');
    cy.server();
    cy.route({
      method: 'GET',
      url: `${geoserverUrl}?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetCapabilities`,
      headers: {
        'Content-type': 'text/xml'
      },
      status: 200,
      response: this.geoserverResponse
    })
      .as('getCapabilities');

Reply:
Tests could be extended by
- handle CORS error
- handle 400, 401, 403, 500, 503 errors
- handle non image response while validating image (clicking button add new layer)